ion-transformation-storage-api/src/main/java/com/connexta/transformation/storage/api/StorageProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api;
+
+public interface StorageProvider {
+
+  /**
+   * Saves the given intel in persistence.
+   *
+   * @param intel the intel to save
+   * @throws StorageException if an error occurs while saving
+   */
+  void save(Intel intel) throws StorageException;
```

Comment:
❗️ Wouldn't this save be for a particular metadata type of a given transform request? 
I would expect to see 2 methods here. One to indicate successful completion where it would expect as argument the transformId, the metadata type string, the content type and an input stream where to retrieve the associated data for the result. The other method should indicate a failure which would expect as arguments the transform id, the metadata type, the reason code and an optional additional message associated with the failure.